Resources/public/js/views/services/plugins/ez-locationsloadplugin.js

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (C) eZ Systems AS. All rights reserved.
+ * For full copyright and license information view LICENSE file distributed with this source code.
+ */
+YUI.add('ez-locationsloadplugin', function (Y) {
+    "use strict";
+    /**
+     * Provides the locations list load plugin
+     *
+     * @module ez-locationsloadplugin
+     */
+    Y.namespace('eZ.Plugin');
+
+    /**
+     *
+     * @namespace eZ.Plugin
+     * @class LocationsLoad
+     * @constructor
+     * @extends eZ.Plugin.ViewServiceBase
+     */
+    Y.eZ.Plugin.LocationsLoad = Y.Base.create('locationsloadplugin', Y.eZ.Plugin.ViewServiceBase, [], {
+
+        initializer: function () {
+            this.onHostEvent('*:loadLocations', this._loadLocations);
+        },
+
+        /**
+         *
+         * @protected
+         * @method _loadLocations
+         * @param {Object} e loadLocations event facade
+         */
+        _loadLocations: function (e) {
+            var options = {api: this.get('host').get('capi')},
+                locations = [];
+
+            e.content.getLocations(options, function (error, response) {
+                if (!error) {
+                    Y.Array.each(response, function (loc) {
+                        var visibility = 'Visible';
+
+                        if (loc.get('hidden')) {
+                            visibility = 'Hidden';
+                        } else if (loc.get('invisible')) {
+                            visibility = 'Invisible';
+                        }
+
+                        locations.push({
+                            'locationId': loc.get('locationId'),
```

Comment:
why do you build an array of plain object ? you can directly give the locations array to the view ?